src/asgi_lifespan/_manager.py

```diff
@@ -81,13 +82,15 @@ async def run_app(self) -> None:
 
             raise
 
-    async def __aenter__(self) -> None:
+    async def __aenter__(self) -> ASGIApp:
         await self._exit_stack.__aenter__()
         await self._exit_stack.enter_async_context(
             self._concurrency_backend.run_in_background(self.run_app)
         )
         try:
             await self.startup()
+            self.app = self._include_state_middleware(self.app)
```

Comment:
I think we need to wrap `app` in `__init__` directly.

Something like:

```python
self.app = state_middleware(app, self._state)

...

def state_middleware(app, state):
    async def app_with_state(scope, receive, send):
        if scope["type"] == "lifespan":
            scope["state"] = state
        await app(scope, receive, send)

    return app_with_state
```

This is because the spec reads (emphasis mine):

https://asgi.readthedocs.io/en/latest/specs/lifespan.html#lifespan-state

> The server will ensure that a shallow copy of the namespace is passed into each subsequent request/response call into the application.

Which is what we do here — include `state` in subsequent requests/responses.

But there's also this bit:

> ASGI servers that implement this feature will provide state **as part of the lifespan scope**

So `state` should _also_ be present as part of the `"type": "lifespan"` scope, right?

I'm actually not sure why the test passes even though `state` is not present during startup.